src/vmm/src/builder.rs

```diff
@@ -465,6 +465,8 @@ pub fn build_microvm_from_snapshot(
         track_dirty_pages,
         vcpu_count,
     )?;
+    vmm.set_rda_pio_serial()
```

Comment:
This would better be `builder::serial_set_rda()`, rather than a `vmm` member function since it's only called during building from snapshot and not useful anywhere else in the lifetime of vmm.

It should also be the same function called in the same place for all archs since it's needed on all archs. I suggest L#533 in this file (in `build_microvm_from_snapshot()` just after `MMIODeviceManager::restore`).